### PR TITLE
[REVIEW] Upgrade to `arrow-11`

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
This PR upgrades arrow to 11 and `kafka` related dependencies to match with arrow's requirements.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: ci/cpu/build.sh [hunk 1]
```diff
@@ -68,6 +68,9 @@ fi
 gpuci_logger "Check compiler versions"
 python --version
 
+gpuci_conda_retry remove --force pyarrow arrow-cpp openssl
+gpuci_mamba_retry install -y "pyarrow=11" "libarrow=11"
```
Comment (galipremsagar) [inline ci/cpu/build.sh:72]:
```suggestion
gpuci_conda_retry remove --force pyarrow arrow-cpp
gpuci_mamba_retry install -y "pyarrow=11" "libarrow=11"
```

File: ci/gpu/build.sh [hunk 2]
```diff
@@ -91,6 +91,9 @@ if [ "$SOURCE_BRANCH" != "main" ] && [[ "${INSTALL_DASK_MAIN}" == 0 ]]; then
   gpuci_mamba_retry install conda-forge::dask==$DASK_STABLE_VERSION conda-forge::distributed==$DASK_STABLE_VERSION conda-forge::dask-core==$DASK_STABLE_VERSION --force-reinstall
 fi
 
+gpuci_conda_retry remove --force pyarrow arrow-cpp openssl
```
Comment (galipremsagar) [inline ci/gpu/build.sh:94]:
```suggestion
gpuci_conda_retry remove --force pyarrow arrow-cpp
```

File: ci/gpu/java.sh [hunk 3]
```diff
@@ -68,6 +68,9 @@ gpuci_logger "Activate conda env"
 . /opt/conda/etc/profile.d/conda.sh
 conda activate rapids
 
+gpuci_conda_retry remove --force pyarrow arrow-cpp openssl
```
Comment (galipremsagar) [inline ci/gpu/java.sh:71]:
```suggestion
gpuci_conda_retry remove --force pyarrow arrow-cpp
```

File: ci/cpu/build.sh [hunk 4]
```diff
@@ -68,6 +68,9 @@ fi
 gpuci_logger "Check compiler versions"
 python --version
 
+gpuci_conda_retry remove --force pyarrow arrow-cpp
+gpuci_mamba_retry install -y "pyarrow=11" "libarrow=11"
```
Comment (galipremsagar) [inline ci/cpu/build.sh:72]:
```suggestion
gpuci_conda_retry remove --force pyarrow arrow-cpp openssl librdkafka
gpuci_mamba_retry install -y "pyarrow=11" "libarrow=11" "librdkafka=1.7.0"
```

File: conda/recipes/libcudf/conda_build_config.yaml [hunk 5]
```diff
@@ -17,10 +17,10 @@ gtest_version:
...
 libarrow_version:
-  - "=10"
+  - "=11"
 
 dlpack_version:
   - ">=0.5,<0.6.0a0"
 
 librdkafka_version:
-  - ">=1.7.0,<1.8.0a0"
+  - ">=1.9.0,<1.10.0a0"
```
Comment (galipremsagar) [inline conda/recipes/libcudf/conda_build_config.yaml:26]:
```suggestion
  - ">=1.7.0,<1.8.0a0"
```

File: dependencies.yaml [hunk 6]
```diff
@@ -58,15 +58,15 @@ dependencies:
...
           - rmm=23.04.*
           - scikit-build>=0.13.1
       - output_types: conda
         packages:
-          - libarrow=10
+          - libarrow=11
           - c-compiler
           - cxx-compiler
-          - librdkafka=1.7.0
+          - librdkafka=1.9.2
```
Comment (galipremsagar) [inline dependencies.yaml:69]:
```suggestion
          - librdkafka=1.7.0
```

File: conda/environments/all_cuda-118_arch-x86_64.yaml [hunk 7]
```diff
@@ -30,8 +30,8 @@ dependencies:
 - gcc_linux-64=9.*
 - hypothesis
 - ipython
-- libarrow=10
-- librdkafka=1.7.0
+- libarrow=11
+- librdkafka=1.9.2
```
Comment (galipremsagar) [inline conda/environments/all_cuda-118_arch-x86_64.yaml:34]:
```suggestion
- librdkafka=1.7.0
```

File: ci/build_cpp.sh [hunk 8]
```diff
@@ -1,12 +1,15 @@
...
+# Copyright (c) 2022-2023, NVIDIA CORPORATION.
 
 set -euo pipefail
 
 source rapids-env-update
 
 export CMAKE_GENERATOR=Ninja
 
+gpuci_conda_retry remove --force pyarrow arrow-cpp openssl librdkafka
+rapids-mamba-retry install -y "pyarrow=11" "libarrow=11" "librdkafka>=1.9.0,<1.10.0a0""
```
Comment (galipremsagar) [inline ci/build_cpp.sh:11]:
```suggestion
rapids-mamba-retry install -y "pyarrow=11" "libarrow=11" "librdkafka>=1.9.0,<1.10.0a0"
```

**Comment by bdice** [comment]
@galipremsagar Is this still targeted for 23.04? Do we need to finish/merge this before burndown?

**Comment by galipremsagar** [comment]
> @galipremsagar Is this still targeted for 23.04? Do we need to finish/merge this before burndown?

Nope. Pushed to future releases.

File: ci/build_cpp.sh [hunk 9]
```diff
@@ -7,6 +7,9 @@ source rapids-env-update
 
 export CMAKE_GENERATOR=Ninja
 
+gpuci_conda_retry remove --force pyarrow arrow-cpp openssl librdkafka
+rapids-mamba-retry install -y "pyarrow=11" "libarrow=11" "librdkafka>=1.9.0,<1.10.0a0"
+
```
Comment (galipremsagar) [inline ci/build_cpp.sh:12]:
```suggestion
```

**Review by bdice — APPROVED** [review]

**Review by ajschmidt8 — APPROVED** [review]

**Review by msadang — APPROVED** [review]
Version updates look good.

**Pull request merged**